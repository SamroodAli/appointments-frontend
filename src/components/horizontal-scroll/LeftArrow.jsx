import { useContext, useState, useEffect } from 'react';
import { VisibilityContext } from 'react-horizontal-scrolling-menu';
import Arrow from './Arrow';

const LeftArrow = () => {
  const {
    isFirstItemVisible,
    scrollPrev,
    visibleItemsWithoutSeparators,
    initComplete,
  } = useContext(VisibilityContext);

  const [disabled, setDisabled] = useState(
    !initComplete || (initComplete && isFirstItemVisible),
  );
  useEffect(() => {
    // NOTE: detect if whole component visible
    if (visibleItemsWithoutSeparators.length) {
      setDisabled(isFirstItemVisible);
    }
  }, [isFirstItemVisible, visibleItemsWithoutSeparators]);

  return (
    <Arrow disabled={disabled} onClick={() => scrollPrev()} className="arrow-left">
      ‚óÅ
    </Arrow>
  );
};

export default LeftArrow;
