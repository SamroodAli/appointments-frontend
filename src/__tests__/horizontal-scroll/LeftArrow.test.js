import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import LeftArrow from '../../components/horizontal-scroll/LeftArrow';
import Wrapper from '../../testWrapper';

test('LeftArrow snapshot testing', () => {
  const tree = renderer.create(
    <Wrapper>
      <LeftArrow />
    </Wrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
