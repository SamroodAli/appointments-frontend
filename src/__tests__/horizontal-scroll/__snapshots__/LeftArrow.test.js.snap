// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftArrow snapshot testing 1`] = `
<div
  className="arrow-left-container arrow-container"
>
  <button
    className="arrow arrow-left"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    ‚óÅ
  </button>
</div>
`;
